components/private_channel/client_private_channel.cc

```diff
@@ -0,0 +1,64 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <iostream>
+
+#include "base/logging.h"
+
+#include "brave/components/private_channel/client_private_channel.h"
+#include "brave/components/private_channel/rust/ffi/src/private_channel.h"
+#include "brave/components/private_channel/utils.h"
+
+ChallengeArtefacts::ChallengeArtefacts() {}
+ChallengeArtefacts::ChallengeArtefacts(const ChallengeArtefacts& other) =
+    default;
+ChallengeArtefacts::~ChallengeArtefacts() {}
+
+SecondRoundArtefacts::SecondRoundArtefacts() {}
+SecondRoundArtefacts::SecondRoundArtefacts(const SecondRoundArtefacts& other) =
+    default;
+SecondRoundArtefacts::~SecondRoundArtefacts() {}
+
+ChallengeArtefacts ChallengeFirstRound(const char** input,
+                                       int input_size,
+                                       const char* server_pk) {
+  struct ChallengeArtefacts artefacts;
+  auto results = private_channel::start_challenge(input, input_size, server_pk);
```

Comment:
what happens if this returns `ResultChallenge::default()`? It seems like we're missing error handling here

Reply:
cc @antonok-edm